spec/models/admin_dashboard_data_spec.rb

```diff
@@ -3,37 +3,129 @@
 require 'rails_helper'
 
 describe AdminDashboardData do
+  after do
+    AdminDashboardData.reset_problem_checks
+    Discourse.redis.flushdb
+  end
+
+  describe "#fetch_problems" do
+    describe "adding problem messages" do
+      it "adds the message and returns it when the problems are fetched" do
+        AdminDashboardData.add_problem_message("dashboard.bad_favicon_url")
+        problems = AdminDashboardData.fetch_problems.map(&:to_s)
+        expect(problems).to include(I18n.t("dashboard.bad_favicon_url", { base_path: Discourse.base_path }))
+      end
+
+      it "does not allow adding of arbitrary problem messages, they must exist in @problem_messages" do
```

Comment:
```suggestion
      it "does not allow adding of arbitrary problem messages, they must exist in AdminDashboardData.problem_messages" do
```

Minor but this should be `@@problem_messages` now and we have a reader method for it.